### PR TITLE
Issue 5098: Rest Server Refactoring (2/4).

### PR DESCRIPTION
**Pull request by co-jo**
**Change log description**  

Factors out the REST server from the `controller` project into a `shared:rest` project. While consumable by any other project, the feature driving this change is integration of the `segmentstore` project with a [health checking](https://github.com/pravega/pravega/pull/5424) service.

**Purpose of the change**  

Logically part of the work necessary to accomplish #5098.

**What the code does**  

Abstracts out any `controller` specific notions and relocates the (slightly) abstracted code into a `shared:rest` project.  

**How to verify it**  
(Optional: steps to verify that the changes are effective)

### CODE REVIEW

File: shared/rest/src/main/java/io/pravega/shared/rest/security/AuthHandlerManager.java [hunk 1]
```diff
@@ -170,7 +183,9 @@ public void registerInterceptors(ServerBuilder<?> builder) {
                                 continue;
                             }
                         }
-                        builder.intercept(new AuthInterceptor(handler));
+                        if (Objects.nonNull(builder)) {
```
Comment (andreipaduroiu) [inline shared/rest/src/main/java/io/pravega/shared/rest/security/AuthHandlerManager.java:186]: `if (builder != null)`

File: shared/rest/src/main/java/io/pravega/shared/rest/RESTApplication.java [hunk 2]
```diff
@@ -13,18 +13,18 @@
...
 
 /**
  * Application to register the REST resource classes.
  */
-public class ControllerApplication extends Application {
+public class RESTApplication extends Application {
     private final Set<Object> resource;
 
-    public ControllerApplication(final Set<Object> resources) {
+    public RESTApplication(final Set<Object> resources) {
```
[the remark this answers is not in the document]
Comment (co-jo) [reply shared/rest/src/main/java/io/pravega/shared/rest/RESTApplication.java:27]: The `Application` superclass requires a `Set`.

File: shared/rest/src/main/java/io/pravega/shared/rest/impl/RESTServerConfigImpl.java [hunk 3]
```diff
@@ -62,4 +64,16 @@ public String toString() {
...
+
+    @Override
+    public boolean isAuthorizationEnabled() {
+        return this.tlsEnabled;
+    }
+
+    @Override
+    public Properties toAuthHandlerProperties() {
+        Properties props = new Properties();
+        // Props to include?
```
[the remark this answers is not in the document]
Comment (co-jo) [reply shared/rest/src/main/java/io/pravega/shared/rest/impl/RESTServerConfigImpl.java:76]: No, that reminder no longer applies.

**Comment by pbelgundi** [comment]
Why is this change necessary for supporting the Health Check framework is not clear to me?

**Comment by co-jo** [comment]
> Why is this change necessary for supporting the Health Check framework is not clear to me?

The segmentstore will be using REST to expose Health details.

**Review by pbelgundi — APPROVED** [review]
I see some failing Unit and Integration tests which need to be fixed, otherwise LGTM.
Lets also make sure that System Tests pass.

**Review by andreipaduroiu — APPROVED** [review]

**Pull request merged**